packages/dd-trace/test/plugins/plugin-structure.spec.js

```diff
@@ -0,0 +1,104 @@
+'use strict'
+
+require('../setup/tap')
+
+const { expect } = require('chai')
+
+const fs = require('fs')
+const path = require('path')
+
+const abstractPlugins = [
+  'web' // web is an abstract plugin, and will not have an instrumentation file
+]
+
+describe('Plugin Structure Validation', () => {
+  const packagesDir = path.join(__dirname, '..', '..', '..')
+  const instrumentationsDir = path.join(packagesDir, 'datadog-instrumentations', 'src')
+
+  let pluginDirs
+  let allPluginIds
+  let instrumentationFiles
+
+  before(() => {
+    pluginDirs = fs.readdirSync(packagesDir)
+      .filter(dir => dir.startsWith('datadog-plugin-'))
+      .filter(dir => {
+        const pluginPath = path.join(packagesDir, dir, 'src', 'index.js')
+        return fs.existsSync(pluginPath)
+      })
+
+    allPluginIds = new Set()
+    instrumentationFiles = new Set(
+      fs.readdirSync(instrumentationsDir)
+        .filter(file => file.endsWith('.js'))
+        .map(file => file.replace('.js', ''))
+    )
```

Comment:
I think it would be best to use `Object.keys(require('packages/datadog-instrumentations/src/helpers/hooks.js'))`

That way we make sure the name of these is actually aligned, since we also use those names later on and they are currently defined manually.

Reply:
We can't do exactly this change, since the hook key corresponds to the module, and our plugin name is a different pattern. Instead, I have added a new test case, that compares the normalized name for the hook key against the normalized name for the plugin name. This should ensure consistency in that same way that we normalize when registering the hooks.